src/utils/format.ts

```diff
@@ -15,27 +28,170 @@ function _getLocaleSymbols(): { thousands: string; decimals: string } {
+  thousandsSymbol: string = THOUSANDS_SYMBOL,
+  decimalSymbol: string = DECIMALS_SYMBOL,
+  smallLimitAsWei: BN,
+  precision: number = DEFAULT_LARGE_NUMBER_PRECISION,
+): string {
+  // Is fraction
+  if (integerPart.isZero()) {
+    // if amount < 1 and fraction < smallLimit (both compared as Wei)
+    // return `< ${smallLimit}`
+    // else return decimals as is
+    // first we need to convert decimals to WEI in order to compare small values
+    const ourDecimalsAsBNWei = new BN(toWei('0.' + decimalsPadded))
+    return ourDecimalsAsBNWei.lt(smallLimitAsWei) ? `< ${_formatDecimalsFromWeiForDisplay(smallLimitAsWei, decimalSymbol)}` : _formatDecimalsFromWeiForDisplay(ourDecimalsAsBNWei)
+  }
+
+  const compactionOptions = { precision, thousandsSymbol, decimalSymbol }
+
+  // Number compacting logic
+  // Anything > 1,000,000,000,000 denoted as <XXX,XXX.xxx>T
+  if (integerPart.gte(BN_1T)) {
+    return _compactLargeBNNumberToString(integerPart, BN_1T, compactionOptions) + 'T'
+  }
+  // Anything not above 1T but > 1,000,000,000 denoted as <XXX.xxx>B
+  if (integerPart.gte(BN_1B)) {
+    return _compactLargeBNNumberToString(integerPart, BN_1B, compactionOptions) + 'B'
+  }
+
+  // At this point can just return thousands separated formatted integer
+  // if decimals are zero
+  if (decimalPart.isZero()) return _formatNumber(integerPart.toString())
+
+  let finalPrecision: number
+
+  // normal format + separator
+  if (integerPart.lt(BN_100K)) {
+    // 99,999.3424
+    finalPrecision = 4
+  } else if (integerPart.lt(BN_1M)) {
+    // 999,999.342
+    finalPrecision = 3
+  } else if (integerPart.lt(BN_10M)) {
+    // 9,999,999.34
+    finalPrecision = 2
+  } else {
+    // 99,999,999.3
+    finalPrecision = 1
+  }
+
+  const amountBeforePrecisionCheck = _formatNumber(integerPart.toString()) + decimalSymbol + decimalsPadded
+  return adjustPrecision(amountBeforePrecisionCheck, finalPrecision).replace(/0+$/, '')
+}
+
+function _decomposeBn(amount: BN, amountPrecision: number, decimals: number): { integerPart: BN; decimalPart: BN, decimalsPadded: string } {
   if (decimals > amountPrecision) {
     throw new Error('The decimals cannot be bigger than the precision')
   }
-  const amountRaw = amount.divRound(TEN.pow(new BN(amountPrecision - decimals)))
-  const integerPart = amountRaw.div(TEN.pow(new BN(decimals))) // 165000 / 10000 = 16
-  const decimalPart = amountRaw.mod(TEN.pow(new BN(decimals))) // 165000 % 10000 = 5000
   // Discard the decimals we don't need
   //  i.e. for WETH (precision=18, decimals=4) --> amount / 1e14
   //        1, 18:  16.5*1e18 ---> 165000
-  return { integerPart, decimalPart }
+  const amountRaw = amount.divRound(TEN.pow(new BN(amountPrecision - decimals)))
+  const integerPart = amountRaw.div(TEN.pow(new BN(decimals))) // 165000 / 10000 = 16
+  const decimalPart = amountRaw.mod(TEN.pow(new BN(decimals))) // 165000 % 10000 = 5000
+  const decimalsPadded = decimalPart.toString().padStart(decimals, '0')
```

Comment:
why duplicating this infor?

Also, I think every `toString()` should have the `toString(10)` parameter?

Reply:
i thought base 10 was default? can add

Reply:
I am adding it here as i use it in several places

Reply:
I'm not sure if it's only for BigNumber, but I remember some issues we had regarding this, cause it was using the scientific notation for large numbers

Reply:
the native `.toString()` uses `10` as default but maybe `BN`/`BigNumber` doesn't. Makes sense to keep it consistent though

Reply:
It does use base 10 by default, but when base is not passed as parameter, it defaults to exponential notation depending on the precision https://mikemcl.github.io/bignumber.js/#toS